src/ai/backend/manager/api/vfolder.py

```diff
@@ -82,6 +84,60 @@
 VFolderRow = Mapping[str, Any]
 
 
+def vfolder_status_check(allowed_status: VFolderOperationStatus):
+    """
+    Checks if the target vfolder status is READY.
+    The decorated handler should prevent user access
+    while storage-proxy operations such as vfolder clone, deletion is handling.
+    """
+
+    def _wrapper(handler: Callable[..., Awaitable[web.Response]]):
+        @functools.wraps(handler)
+        async def _wrapped(request: web.Request, *args, **kwargs) -> web.Response:
+            root_ctx: RootContext = request.app["_root.context"]
+            domain_name = request["user"]["domain_name"]
+            user_role = request["user"]["role"]
+            user_uuid = request["user"]["uuid"]
+            allowed_vfolder_types = await root_ctx.shared_config.get_vfolder_types()
+            folder_name = request.match_info["name"]
+            vf_name_conds = vfolders.c.name == folder_name
+            if allowed_status == VFolderOperationStatus.READY:
+                vf_status_conds = vfolders.c.status.in_(
+                    [
+                        VFolderOperationStatus.READY,
+                        VFolderOperationStatus.PREPARED,
+                        VFolderOperationStatus.MOUNTED,
+                        VFolderOperationStatus.CLONING,
+                        VFolderOperationStatus.DELETING,
+                    ]
+                )
+            elif allowed_status == VFolderOperationStatus.PREPARED:
+                vf_status_conds = vfolders.c.status.in_(
+                    [
+                        VFolderOperationStatus.READY,
+                        VFolderOperationStatus.PREPARED,
+                    ]
+                )
```

Comment:
vfolders with any status are filtered if `VFolderOperationStatus.READY` is given?
What does `allowed_status` mean?

Reply:
`allowed_status` is a parameter which is used to define vfolder statuses (values listed in `vf_status_conds`) to determine whether to lock the vfolder operation or not.

It seems not intuitive, so I modified it to `perm` and listed all cases of `perm`(`VFolderOperationStatus`) by using elif statements.